nemo/collections/common/tokenizers/japanese_tokenizers.py

```diff
@@ -0,0 +1,54 @@
+# Copyright (c) 2021, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import List
+
+from sacremoses import MosesDetokenizer, MosesTokenizer
+
+from nemo.collections.common.tokenizers.sentencepiece_detokenizer import SentencePieceDetokenizer
+from nemo.collections.common.tokenizers.sentencepiece_tokenizer import SentencePieceTokenizer
+
+
+class JapaneseDetokenizer:
+    def __init__(self):
+        self.moses_detokenizer = MosesDetokenizer(lang='ja')
+        self.sp_detokenizer = SentencePieceDetokenizer()
+
+    def detokenize(self, tokens: List[str]) -> str:
+        """
+        Detokenizes a list of sentencepiece tokens in Japanese
+        Args:
+            tokens: list of strings as tokens
+        Returns:
+            detokenized Japanese string
+        """
+        text = self.sp_detokenizer.detokenize(tokens)
+        return self.moses_detokenizer.detokenize(text)
+
+
+class JapaneseTokenizer:
+    def __init__(self, sp_tokenizer_model):
+        self.moses_tokenizer = MosesTokenizer(lang='ja')
+        self.sp_tokenizer = SentencePieceTokenizer(model_path=sp_tokenizer_model)
```

Comment:
can you make sure to say that `sp_tokenizer_model` has to be a string to a sentencepiece model, not an actual sentencepiece object?

Reply:
Sure, I've added a docstring for the class and changed the variable name to `sp_tokenizer_model_path`. Thanks!